identity-service/src/routes/wormhole.js

```diff
@@ -0,0 +1,137 @@
+}
+
+const getTxProps = (senderAddress, method) => {
+  const sha = crypto.createHash('sha256').update(JSON.stringify({ senderAddress, ...method })).digest('hex')
+  return {
+    sha,
+    txProps: {
+      contractAddress: method.contractAddress,
+      encodedABI: method.encodedABI,
+      senderAddress: senderAddress,
+      gasLimit: method.gasLimit || null
+    }
+  }
+}
+
+const relayWormhole = async (
+  req,
+  audiusLibs,
+  senderAddress,
+  permit,
+  transferTokens,
+  reportError
+) => {
+  const logs = []
+  const context = {}
+  try {
+    logs.push(`Attempting Permit for sender: ${senderAddress}`)
+    const { sha: permitSHA, txProps: permitTxProps } = getTxProps(senderAddress, permit)
+    const permitTxResponse = await ethTxRelay.sendEthTransaction(req, permitTxProps, permitSHA)
+    logs.push(`Permit Succeded with tx hash: ${permitTxResponse.txHash}`)
+    context.permitTxHash = permitTxResponse.txHash
+
+    // Send off transfer tokens to eth wormhole tx
+    logs.push(`Attempting Transfer Tokens for sender: ${senderAddress}`)
+    const { sha: transferTokensSHA, txProps: transferTokensTxProps } = getTxProps(senderAddress, transferTokens)
+    const estimatedGas = await ethTxRelay.estimateEthTransactionGas(senderAddress, transferTokensTxProps.contractAddress, transferTokensTxProps.encodedABI)
+    const gasMultiplier = 1.5
+    transferTokensTxProps.gasLimit = Math.floor(estimatedGas * gasMultiplier)
+
+    const transferTokensTxResponse = await ethTxRelay.sendEthTransaction(req, transferTokensTxProps, transferTokensSHA)
+    const transferTxHash = transferTokensTxResponse.txHash
+    context.transferTxHash = transferTxHash
+    logs.push(`Attempting Transfer Tokens for sender: ${transferTxHash}`)
+    const feePayerAccount = getFeePayer()
+
+    const signTransaction = async (transaction) => {
+      transaction.partialSign(feePayerAccount)
+      return transaction
+    }
+
+    // Gather VAA attestations, submit to solana and realize the funds at the taret address
+    const response = await audiusLibs.wormholeClient.attestAndCompleteTransferEthToSol(transferTxHash, signTransaction, {
+      transport: NodeHttpTransport()
+    })
+    if (response.error) {
+      logs.push(`Attest and complete transfer from eth to sol failed on phase: ${response.phase} \n error: ${response.error} \n with logs: ${response.logs.join(',')}`)
+      const errorMessage = response.error.toString()
+      await reportError({ logs: logs.join(','), error: errorMessage })
+      return { error: errorMessage }
+    }
+
+    logs.push(`Attest and complete transfer from eth to sol succeeded: ${response.transactionSignature}`)
+    context.completeTransferSignature = response.transactionSignature
+
+    logs.push(`Attest and complete transfer from eth to sol succeeded: ${response.transactionSignature}`)
+    context.completeTransferSignature = response.transactionSignature
+
+    req.logger.info(context, logs.join(','))
```

Comment:
do we want to make sure we still log if something throws and we end up in the catch block?